### PR TITLE
tlrc: init at 1.7.1

### PR DESCRIPTION
**Pull request by acuteenvy**
## Description of changes

Added [tlrc](https://github.com/tldr-pages/tlrc)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by IogaMaster — APPROVED** [review]

**Review by ambroisie — APPROVED** [review]

File: pkgs/by-name/tl/tlrc/package.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  postInstall = ''
+    installManPage tldr.1
+
+    installShellCompletion \
+      --name tldr --bash $releaseDir/build/tlrc-*/out/tldr.bash \
+      --zsh $releaseDir/build/tlrc-*/out/_tldr \
+      --fish $releaseDir/build/tlrc-*/out/tldr.fish
```
Comment (ambroisie) [inline pkgs/by-name/tl/tlrc/package.nix:28]: Formatting nitpick, as aligning the arguments makes it more readable:

```suggestion
    installShellCompletion --name tldr \
      --bash $releaseDir/build/tlrc-*/out/tldr.bash \
      --zsh $releaseDir/build/tlrc-*/out/_tldr \
      --fish $releaseDir/build/tlrc-*/out/tldr.fish
```

**Review by IogaMaster — APPROVED** [review]
LGTM

**Review by kbdharun — APPROVED** [review]

File: pkgs/by-name/tl/tlrc/package.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+, installShellFiles
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "tlrc";
+  version = "1.7.1";
+
+  src = fetchFromGitHub {
+    owner = "tldr-pages";
+    repo = pname;
```
Comment (beeb) [inline pkgs/by-name/tl/tlrc/package.nix:13]:
```suggestion
    repo = "tlrc";
```

Seems to be required to not use `pname` here. Not sure why but I got told so on previous PRs

**Review by beeb — CHANGES_REQUESTED** [review]
Minor change, otherwise similar to what I would have done.

Can test on darwin x86 later today if it doesn't get merged before then.

File: pkgs/by-name/tl/tlrc/package.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      --bash $releaseDir/build/tlrc-*/out/tldr.bash \
+      --zsh $releaseDir/build/tlrc-*/out/_tldr \
+      --fish $releaseDir/build/tlrc-*/out/tldr.fish
+  '';
+
+  meta = with lib; {
+    description = "Official tldr client written in Rust";
+    homepage = "https://github.com/tldr-pages/tlrc";
+    changelog = "https://github.com/tldr-pages/tlrc/releases/tag/v${version}";
+    license = licenses.mit;
```
Comment (beeb) [inline pkgs/by-name/tl/tlrc/package.nix:35]:
```suggestion
    license = with licenses; [ mit ];
```

**Review by beeb — CHANGES_REQUESTED** [review]
Oops I forgot this change request

File: pkgs/by-name/tl/tlrc/package.nix (hunk 3 above)
Comment (acuteenvy) [reply pkgs/by-name/tl/tlrc/package.nix:35]: There are 17 packages using `license = with licenses; [ mit ]` and 5412 packages using `license = licenses.mit`. Are you sure this is preferred?

Comment (beeb) [reply pkgs/by-name/tl/tlrc/package.nix:35]: Not sure but it makes it easier to edit if there are multiple licenses down the line. I won't be mad if you don't change it, just my preference ;)

**Review by beeb — APPROVED** [review]
Tested on darwin x86_64

**Review by kbdharun — APPROVED** [review]

**Pull request merged**